File: classes/parte-9-ingenieria-de-software-poliglota/150-refactorizacion-segura/implementaciones/c/main.c
#include <stdio.h>

int main(void) {
    long n;
    if (scanf("%ld", &n) != 1) return 1;
    long viejo = n * 2, nuevo = n + n;
    printf("equivalente=%s resultado=%ld\n", viejo == nuevo ? "true" : "false", nuevo);
    return 0;
}

File: classes/parte-9-ingenieria-de-software-poliglota/150-refactorizacion-segura/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	n, _ := strconv.ParseInt(strings.TrimSpace(line), 10, 64)
	viejo, nuevo := n*2, n+n
	res := "false"
	if viejo == nuevo {
		res = "true"
	}
	fmt.Printf("equivalente=%s resultado=%d\n", res, nuevo)
}
